### PR TITLE
chore(dm): add tracking plan stat per event level

### PR DESCRIPTION
**Pull request by abhimanyubabbar**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/services/trackingPlan.ts [hunk 1]
```diff
@@ -5,71 +5,85 @@ import eventValidator from '../util/eventValidation';
...
+          statusCode: 200,
+          validationErrors: failure['validationErrors'],
+          error: JSON.stringify(constructValidationErrors(failure['validationErrors'])),
+        };
+
       } catch (error) {
+        exception = true;
+
         const errMessage = `Error occurred while validating : ${error}`;
         logger.error(errMessage);
```
Comment (Jayachand) [inline src/services/trackingPlan.ts:49]: add event name in logger to pin point where error happened

File: src/services/trackingPlan.ts [hunk 2]
```diff
@@ -5,71 +5,85 @@ import eventValidator from '../util/eventValidation';
...
+
+      // stat which is fired per tp event validation level
+      // containing labels which allows us to identify the information
+      // at each stage.
+      stats.counter('tp_event_validation', 1, {
+        ...metaTags,
+        workspaceId: event.metadata?.workspaceId,
+        trackingPlanId: event.metadata?.trackingPlanId,
+        status: errorReport?.status,
+        violationType: errorReport.violationType ? errorReport.violationType : '',
```
Comment (Jayachand) [inline src/services/trackingPlan.ts:74]: one event can have multiple violations. We can stop capturing this info

File: package-lock.json [hunk 3]
```diff
@@ -8158,6 +8157,7 @@
     },
     "node_modules/available-typed-arrays": {
       "version": "1.0.5",
+      "dev": true,
```
Comment (Jayachand) [inline package-lock.json:8160]: remove changes from package-lock file

File: src/services/trackingPlan.ts [hunk 4]
```diff
@@ -5,71 +5,85 @@ import eventValidator from '../util/eventValidation';
 import stats from '../util/stats';
 
 export default class TrackingPlanservice {
-  public static async validateTrackingPlan(events, requestSize, reqParams) {
+  /**
+   * validate takes in events as argument and iterates over the events
+   * validating them using the tracking
```
Comment (Jayachand) [inline src/services/trackingPlan.ts:10]:
```suggestion
   * validating them using the tracking plan
```

File: src/services/trackingPlan.ts [hunk 5]
```diff
@@ -5,71 +5,85 @@ import eventValidator from '../util/eventValidation';
...
-            metadata: event.metadata,
-            statusCode: 200,
-            validationErrors: hv['validationErrors'],
-            error: JSON.stringify(constructValidationErrors(hv['validationErrors'])),
-          });
-          stats.counter('tp_propagated_events', 1, {
-            ...metaTags,
-          });
-        }
+        const failure = await eventValidator.handleValidation(parsedEvent);
```
Comment (Jayachand) [inline src/services/trackingPlan.ts:30]:
```suggestion
        const validatedEvent = await eventValidator.handleValidation(parsedEvent);
```

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

**Pull request closed without merging**